src/libraries/System.Console/src/System/IO/StdInReader.cs

```diff
@@ -79,129 +80,184 @@ internal unsafe int ReadStdin(byte* buffer, int bufferSize)
 
         public override string? ReadLine()
         {
-            return ReadLine(consumeKeys: true);
+            return ReadLine(consumeKeys: true).line;
         }
 
-        private string? ReadLine(bool consumeKeys)
+        public int ReadLine(byte[] buffer, int offset, int count)
         {
+            return ReadLine(consumeKeys: true, buffer, offset, count).bytesRead;
+        }
+
+        private (string? line, int bytesRead) ReadLine(bool consumeKeys, byte[]? buffer = null, int offset = 0, int count = 0)
```

Comment:
it: this feels like two separate methods that have been merged into one. Do you think it might make sense to refactor somewhat (e.g. by having a "Core" method that simply populates the StringBuilder).